### DISCUSSION TITLE
Typecript for UseControllerProps

### DISCUSSION THREAD

**Discussion by anshour** [react-hook-form/react-hook-form#8958] [Q&A] [answered]
Hi, i wanna build some reusable input that always integrated with react hook form, because i dont want to always register useform hooks to my input component. everything works, except the typescript compiler keep telling me that "Control is not assignable". how to fix it?

https://codesandbox.io/s/reusable-react-hook-form-wqrdli?file=/src/App.tsx

**Comment by Moshyfawn** [comment, marked as answer]
`UseControllerProps` accepts `FieldValues` generic. Additionally, `useForm` expects a `FieldValues` generic to have `control` infer it's types.

Check out [this working csb](https://codesandbox.io/s/reusable-react-hook-form-forked-nviqzc?file=%2Fcomponents%2Fform%2FSmartInput.tsx%3A854-858)

**Reply by anshour** [reply]
Wow, it works. thanks
